### DISCUSSION TITLE
Implementing Password Show/Hide functionality

### DISCUSSION THREAD

**Discussion by n-srinidhi** [h2oai/wave#1976] [Q&A] [answered]
I'm trying to implement the password show/hide option available in most login forms. Since the icon attribute of a textbox isnt clickable, is there any built in functionality or a workaround to implement it?

![image](https://github.com/h2oai/wave/assets/117831190/022edfe1-8ef1-4215-bd72-9ab36787710f)

`ui.textbox(name='login_pass',placeholder='Password',password=True,required=True,width='250px',icon='View'),`

**Comment by marek-mihok** [comment, marked as answer]
@n-srinidhi, there is no built-in functionality right now, but you can use an icon button next to the textfield to toggle the password visibility.
```py
from h2o_wave import main, app, Q, ui

password_hidden = True


@app('/demo')
async def serve(q: Q):
    global password_hidden
    if not q.client.initialized:
        q.page['meta'] = ui.meta_card(
            box='',
            layouts=[
                ui.layout(
                    breakpoint='xs',
                    zones=[ui.zone(name='main', size='1')])]
        )
        q.page['login_form'] = ui.form_card(
            box='main',
            items=[
                ui.inline(
                    direction='column',
                    items=[
                        ui.text(content='Login', size='l', name='login_label'),
                        ui.textbox(name='login_id', placeholder='Enter Login ID', required=True, width='240px'),
                        ui.inline(
                            items=[
                                ui.textbox(name='login_pass', placeholder='Password', password=password_hidden,
                                           required=True, width='200px'),
                                ui.button(name='toggle_password', icon='RedEye')
                            ])
                    ],
                    align='center',
                    justify='center',
                    height='1'
                ),
            ]
        )
        q.client.initialized = True
    elif q.args.toggle_password:
        password_hidden = not password_hidden
        q.page['login_form'].login_pass.password = password_hidden
        q.page['login_form'].toggle_password.icon = 'RedEye' if password_hidden else 'Hide'
    await q.page.save()

```

https://github.com/h2oai/wave/assets/23740173/7a87cb5e-e428-42b3-b335-031403b1ab8e

However if you need a built-in functionality, feel free to open a [new feature request](https://github.com/h2oai/wave/issues/new?assignees=&labels=feature&projects=&template=feature_request.md&title=). It should be doable since [FluentUI's Textfield](https://developer.microsoft.com/en-us/fluentui#/controls/web/textfield) supports `canRevealPassword` prop which handles the functionality (including the show/hide icons) for you.

**Reply by n-srinidhi** [reply]
@marek-mihok  Thank you!
